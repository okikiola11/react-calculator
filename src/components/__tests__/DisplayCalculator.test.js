import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import DisplayCalculator from '../DisplayCalculator';

describe('Display Calculator', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DisplayCalculator />, div);
  });

  it('renders display calculator component correctly', () => {
    render(<DisplayCalculator result="9" />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<DisplayCalculator result="9" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
