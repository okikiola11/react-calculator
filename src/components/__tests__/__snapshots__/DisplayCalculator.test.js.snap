// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Calculator matches snapshot 1`] = `
<section
  className="sc-bdnxRM hLccjP"
>
  <div
    className="display"
  >
    <h1>
      9
    </h1>
  </div>
</section>
`;
